### PR TITLE
feat(portal): header & footer links are now retrieved from portal API

### PR DESCRIPTION
**Pull request by phiz71**
Closes gravitee-io/issues#3048

### CODE REVIEW

File: src/app/layouts/layout/layout.component.ts [hunk 1]
```diff
@@ -87,47 +89,51 @@ export class LayoutComponent implements OnInit {
...
-        { title: i18n('footer.links.status'), link: '/' },
-      ]
-    };
+    this.portalService.getPortalLinks().subscribe((portalLinks) => {
+      if (portalLinks.slots) {
+        if (portalLinks.slots.header && portalLinks.slots.header[0] && portalLinks.slots.header[0].links ) {
+          this.links.header = this._buildLinks(portalLinks.slots.header[0].links);
+        }
+        if (portalLinks.slots.subfooter && portalLinks.slots.subfooter[0] && portalLinks.slots.subfooter[0].links ) {
+          this.links.subfooter = this._buildLinks(portalLinks.slots.subfooter[0].links);
```
Comment (gcusnieux) [inline src/app/layouts/layout/layout.component.ts]: you could pass the slot directly to the function `_buildLinks( { links } )`

File: src/app/pages/singlepage/singlepage.component.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ app-gv-page {
+    display: block;
```
Comment (gcusnieux) [inline src/app/pages/singlepage/singlepage.component.css]: maybe it should be applied every time?

File: src/app/app.module.ts [hunk 3]
```diff
@@ -105,6 +106,7 @@ import { AuthService } from './services/auth.service';
     GvSearchComponent,
     UserAvatarComponent,
     GvContactComponent,
+    SinglepageComponent,
```
[the remark this answers is not in the document]
Comment (phiz71) [reply src/app/app.module.ts]: Done

File: src/app/layouts/layout/layout.component.html [hunk 4]
```diff
@@ -57,19 +59,19 @@ <h1 class="layout__header__homepage__title">{{'homepage.title' | translate}}</h
       <div class="w-20 fl">
         <a [routerLink]="'/'" class="layout__footer__top__logo" title="{{ 'site.title' | translate }}"></a>
       </div>
-      <div class="layout__footer__top__category" *ngFor="let category of links.categorized">
+      <div class="layout__footer__top__category" *ngFor="let category of links.footer">
         <span class="layout__footer__top__category__title">{{category.title | translate}}</span>
         <ul class="layout__footer__top__category__links">
           <li *ngFor="let link of category.links">
-            <a [href]="link.link">{{link.title | translate}}</a>
+              <gv-link [path]="link.link" [target]="link.target" title="{{ link.title | translate }}"></gv-link>
```
[the remark this answers is not in the document]
Comment (phiz71) [reply src/app/layouts/layout/layout.component.html]: Done

**Pull request merged**